File: frontend/src/types/job.ts
export interface Job {
  id: string;
  source_url: string;
  company_name: string | null;
  job_title: string | null;
  location: string | null;
  salary_range: string | null;
  description_text: string | null;
  requirements: Record<string, unknown> | null;
  ats_platform: string | null;
  scrape_status: string;
  scrape_error: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/job.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class JobCreate(BaseModel):
    source_url: str


class JobResponse(BaseModel):
    id: UUID
    source_url: str
    company_name: Optional[str] = None
    job_title: Optional[str] = None
    location: Optional[str] = None
    salary_range: Optional[str] = None
    description_text: Optional[str] = None
    requirements: Optional[dict] = None
    ats_platform: Optional[str] = None
    scrape_status: str
    scrape_error: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
